### ISSUE TITLE
lib-css .lib-message-icon-inner does not vertically align icon in the middle. Should it? Can it?

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#31012]
### Preconditions (*)

1. magento 2.4-develop
1. Magento 2.4.1

### Steps to reproduce (*)

1. Add a page containing the following html

```
<div class="message info">
<div>This is a really long message that should span multiple line. Oh I've ran out of stuff to say, so perhaps I'll let you know that a black cat is outside my window meowing, presumably because I've fed it cat treats before and he remebers. Woo, an amazon delivery has just arrived. Thats anough.</div>
</div>
```

### Expected result (*)

1. The icon should be vertically aligned in the center.
![image](https://user-images.githubusercontent.com/30697781/99953647-790b2b80-2d79-11eb-854d-55cdb07135b1.png)

### Actual result (*)

1. The icon is aligned at the top.
![image](https://user-images.githubusercontent.com/30697781/99953180-cd61db80-2d78-11eb-99fe-ddbede8fe588.png)

There has obviously been an attempt to centrally align the icon as the css shows, but the vertical-align: middle attribute is not being applied as there is a conflict with the absolute positioning being applied. So, is it possible to configure the icon so it is vertically aligned. Not at present I think.

![image](https://user-images.githubusercontent.com/30697781/99953328-0601b500-2d79-11eb-8a08-5377960ee7bc.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by gwharton** [comment]
I can achieve vertical aligning in my theme using

```
    .message.info {
        .lib-message-icon-inner(info);
        > *:first-child:before {
            .lib-css(margin, 0);
            .lib-css(top, 50%);
            .lib-css(transform, translateY(-50%));
        }
    }
```
But should libcss support vertically aligning the icon through inbuilt mixins?

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-39299` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by mrtuvn** [comment]
if notices/warning with multiple line and long text i think it's fine. Ok it's can be vertical middle but imo it don't have to be vertical. I can show like default is fine 
If we have warning/error with one line text ! icon will auto vertical align in one line

**Comment by mrtuvn** [comment]
Can you create PR for this ?

**Comment by gwharton** [comment]
I'm not really sure what the solution should be. It feels like the existing mixins should support center alignment somehow. This is from https://magento-devdocs.github.io/magento2-ui-library/messages.html

```
@message-icon__top	18px	'' | false | value	Message icon top position
@message-icon__bottom	false	'' | false | value	Message icon bottom position

```
I notice that top and bottom can both be set to false. In this case, wouldn't it make sense for it to center itself? At the moment if top and bottom are set to false, it goes to the VERY top.

![image](https://user-images.githubusercontent.com/30697781/100008825-daf18280-2dc5-11eb-8c0e-ffd72f576891.png)

**Comment by gwharton** [comment]
Oh hang on. I've got it.

@message-icon__top: 50%;
@message-icon__bottom: false;

Because the libcss applies a top margin of -(font-size/2) to the icon, it is shifted up by half its height, so by setting top to 50%, it centers it.

**Comment by gwharton** [comment]
I'll close this issue as the existing mixins DO actually support centering by setting top to 50%.

**Comment by magento-engcom-team** [comment]
Hi @gwharton. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-39299` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.2` release.
